crates/query-engine/metadata/src/metadata/database.rs

```diff
@@ -2,32 +2,40 @@
 
 use std::collections::{BTreeMap, BTreeSet};
 
-/// Map of all known composite types.
-#[derive(Debug, Clone, PartialEq, Eq, Default)]
-
-pub struct CompositeTypes(pub BTreeMap<String, CompositeType>);
-
-/// A Scalar Type.
+/// A name of a Scalar Type, as it appears in the NDC scheme.
 #[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
 pub struct ScalarTypeName(pub String);
 
+/// The name of a Composite Type, as it appears in the NDC schema
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct CompositeTypeName(pub String);
+
 /// The type of values that a column, field, or argument may take.
 #[derive(Debug, Clone, PartialEq, Eq)]
-
 pub enum Type {
     ScalarType(ScalarTypeName),
-    CompositeType(String),
+    CompositeType(CompositeTypeName),
     ArrayType(Box<Type>),
 }
 
+/// Map of all known composite types.
+#[derive(Debug, Clone, PartialEq, Eq, Default)]
+pub struct CompositeTypes(pub BTreeMap<CompositeTypeName, CompositeType>);
+
+impl CompositeTypes {
```

Comment:
If you are already doing `impl`s on these types, perhaps you'd prefer to impl [Default](https://doc.rust-lang.org/std/default/trait.Default.html)?

Reply:
I don't know that I like that. I don't think Default is right, since there is no way you can generally just take an empty environment and exepect to have a good time.

Reply:
Ok.